src/App.js

```diff
@@ -1,22 +1,229 @@
+
+const http = axios.create()
 
 class App extends Component {
+  state = {
+    selectedLessons,
+    allLessons,
+    isOpen: false
+  }
+
+  handleStateChange = debounce((changes) => {
+    if (changes.hasOwnProperty('inputValue') && !isUndefined(changes['inputValue'])) {
+      return http
+        .get(`http://egghead.af:5000/api/v1/lessons?q=${changes['inputValue']}&size=20&page=1`)
```

Comment:
this requires a [local elasticsearch setup](https://gist.github.com/jpalala/ab3c33dd9ee5a6efbdae). Once that is installed you need to `Lesson.reindex` inside of the rails console (`rails c`). Also assumes this PR is merged: https://github.com/eggheadio/egghead-rails/pull/2228